AppBuilder/platform/dataFields/ABFieldConnect.js

```diff
@@ -627,79 +627,172 @@ module.exports = class ABFieldConnect extends ABFieldConnectCore {
          if (data.map) {
             selectedData = data.map(function (d) {
                // display label in format
-               if (d) d.text = d.text || linkedObject.displayData(d);
+               if (d) {
+                  d.text = d.text || linkedObject.displayData(d);
+                  d.value = d.text;
+               }
 
                return d;
             });
          } else if (data.id || data.uuid) {
             selectedData = data;
             selectedData.text =
                selectedData.text || linkedObject.displayData(selectedData);
+            selectedData.value = selectedData.text;
+         }
+      }
+
+      return selectedData;
+   }
+
+   /**
+    * @method pullRecordRelationValues
+    *
+    * On the Web client, we want our returned relation values to be
+    * ready for Webix objects that require a .text and .value field.
+    *
+    * @param {*} row
+    * @return {array}
+    */
+   pullRecordRelationValues(record) {
+      var selectedData = [];
+
+      var data = record;
+      var linkedObject = this.datasourceLink;
+
+      if (data && linkedObject) {
+         // if this select value is array
+         if (data.map) {
```

Comment:
okay, works for me. I honestly don't like using map because I read it was slower than just a normal loop. So if we want to refactor this later that is great.